coalib/output/ConfWriter.py

```diff
@@ -89,15 +92,69 @@ def __write_key_val(self, keys, val):
 
         # Add escape characters as appropriate
         keys = [escape(key, chain(['\\'],
-                                  self.__key_value_delimiters,
+                                  self.__key_value_define_delimiters,
                                   self.__comment_separators,
                                   self.__key_delimiters,
                                   self.__section_override_delimiters))
                 for key in keys]
         val = escape(val, chain(['\\'], self.__comment_separators))
 
-        self.__file.write((self.__key_delimiter + ' ').join(keys) + ' ' +
-                          self.__key_value_delimiter + ' ' + val + '\n')
+        append_keys = []
+        other_keys = []
+        for key in keys:
+            if (defaults and
+                    (key in defaults and
+                     (val.startswith(str(defaults[key])+',')))):
+                append_keys.append(key)
+            else:
+                other_keys.append(key)
+        self.__write_keys_val_to_file(append_keys, other_keys, val, defaults)
+
+    def __write_keys_val_to_file(self, append_keys, other_keys, val, defaults):
+        if append_keys:
+            append_keys = sorted(append_keys,
+                                 key=lambda key: len(defaults[str(key)]),
+                                 reverse=True)
+            append_keys_iter = append_keys.__iter__()
+            write_keys = []
+            def_val = None
+            try:
+                while True:
```

Comment:
the more I look at it the more I think that a simple for loop would do the job quite well, simplifying the code here a bit